GitHub gradle.yml/graalvm.yml: Add JDK 23

msgilligan opened the pull request:
Try building with JDK 23-ea

This is a child of PR #3442 

I'm expecting this will show a JavaDoc build error that I do not yet understand. (Update: JavaDoc error fixed in PR #3444)

msgilligan commented:
Once PR #3442 and #3444 are merged, this can be rebased and should pass CI.

schildbach commented:
Feel free to rebase.

schildbach commented:
However I fear GitHub Actions is not yet ready for JDK 23:

>  Error: Unable to find the latest Java version for '23-ea'. Please make sure the java-version is set correctly. If you think this is a mistake, please file an issue at: https://github.com/graalvm/setup-graalvm/issues.

schildbach commented:
I'm setting the 0.18 milestone on this, because I think there is no point in tracking alpha-like versions on release branches (which this will become any day now). Instead, I'd merge this early in the 0.18 cycle (if GitHub Actions allows).

msgilligan commented:
> However I fear GitHub Actions is not yet ready for JDK 23:
> 
> > Error: Unable to find the latest Java version for '23-ea'. Please make sure the java-version is set correctly. If you think this is a mistake, please file an issue at: https://github.com/graalvm/setup-graalvm/issues.

The `graalvm/setup-graalvm` action doesn't seem to work with `23-ea`  but the `setup-java` does.

I've rebased and removed the `23-ea` from `graalvm.yml`.

msgilligan commented:
> I'm setting the 0.18 milestone on this

That's fine.

> alpha-like versions 

Well, the latest `23-ea` is the "initial release candidate" with a "final release candidate" expected on August 22nd. In every JDK release that I'm aware of the GA has been "final release candidate" without modification.

In any event, the important thing is that we found and fixed our JavaDoc error and 0.17 should now be compatible with JDK 23.

schildbach commented:
Has the "final release candidate" been released?

schildbach commented:
This would need to be rebased and changed to pull the final release of JDK 23.

schildbach commented:
Merged.
